var/spack/repos/builtin/packages/py-onnxruntime/package.py

```diff
@@ -63,6 +63,8 @@ class PyOnnxruntime(CMakePackage, PythonExtension):
     patch("libiconv-1.10.patch", level=0, when="@1.10.0 ^libiconv")
     # https://github.com/microsoft/onnxruntime/commit/de4089f8cbe0baffe56a363cc3a41595cc8f0809.patch
     patch("gcc11.patch", level=1, when="@1.7.2")
+    # https://github.com/microsoft/onnxruntime/pull/16257
+    patch("stringpiece_1_10.patch", when="@1.10.0")
```

Comment:
This applies through version :1.15 (but not anymore in 1.16.0), so you could expand this for future versions:
```suggestion
    patch("stringpiece_1_10.patch", when="@1.10.0:1.16.0")
```
It is also possible to refer to the upstream patch link, and not upload it here, e.g.
```suggestion
    patch(
        "https://github.com/microsoft/onnxruntime/commit/a3a443c80431c390cbf8855e9c7b2a95d413cd54.patch",
        sha256="459d52008ffcca408baa9193d63eb228927a9aa9aaa08597c813cd8cdacba00e",
        when="@1.10.0:1.16.0",
    )
```
Finally, it may be useful to evaluate if this patch applies to 1.7.2, since that's still an active version in spack too.